drivers/spi/spi_rpi_pico_pio.c

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright (c) 2023 Stephen Boylan <stephen.boylan@beechwoods.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_pico_spi_pio
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_pico_pio);
+
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/pinctrl.h>
+#include "spi_context.h"
+
+#include <zephyr/drivers/misc/pio_rpi_pico/pio_rpi_pico.h>
+
+#include <hardware/pio.h>
+#include "hardware/clocks.h"
+
+#define PIO_CYCLES     (4)
+#define PIO_FIFO_DEPTH (4)
+
+struct spi_pico_pio_config {
+	const struct device *piodev;
+	const struct pinctrl_dev_config *pin_cfg;
+	struct gpio_dt_spec clk_gpio;
+	struct gpio_dt_spec mosi_gpio;
+	struct gpio_dt_spec miso_gpio;
+	const uint32_t clock_freq;
+};
+
+struct spi_pico_pio_data {
+	struct spi_context spi_ctx;
+	uint32_t tx_count;
+	uint32_t rx_count;
+	PIO pio;
+	size_t pio_sm;
+	uint32_t dfs;
+};
+
+RPI_PICO_PIO_DEFINE_PROGRAM(spi_cpol_0_cpha_0, 0, 1,
+			    /*     .wrap_target */
```

Comment:
Yes - that formatting should match the output of pioasm, with "//" comments replaced with "/* */".

Reply:
Follow-up on this issue:  The comments only align when displayed with 1 tab == 4 spaces.  Since the Zephyr rules require using tabs, not spaces, the alignment of the comments will depend on the tool used to view the source.

Reply:
The code base is done with 8 spaces, Linux style, fairly common for C.

Reply:
Mandatory Linux coding-style reference: https://elixir.bootlin.com/linux/latest/source/Documentation/process/coding-style.rst#L21

Reply:
@fabiobaltieri - This is why they invented spaces.  :-)

Reply:
> @fabiobaltieri - This is why they invented spaces. :-)

Thanks, I'll have 8 of them please. :-)